tests/serac_sets.cpp

```diff
@@ -0,0 +1,296 @@
+      *pmesh, FiniteElementState::Options{.order      = 0,
+                                          .vector_dim = 1,
+                                          .coll = {std::make_unique<mfem::L2_FECollection>(0, pmesh->SpaceDimension())},
+                                          .ordering = mfem::Ordering::byNODES,
+                                          .name     = "stars",
+                                          .alloc_gf = true});
+  stars.project(stars_coef);
+
+  auto stars_attr_set =
+      Set(mfem_ext::MakeAttributeList<std::vector<int>>(*pmesh, stars_coef, mfem_ext::digitize::floor));
+
+  // Set two additional bdr attributes: 1 = everything, 2 = bottom, 3 = right side
+  auto bdr_assign_coef = mfem::FunctionCoefficient([&](const mfem::Vector& x) {
+    if (fabs(x[1] - 0.) < 1.e-5) {
+      return 2.;
+    } else if (fabs(x[0] - flag_width) < 1.e-5) {
+      return 3.;
+    }
+    return 1.;
+  });
+
+  mfem_ext::AssignMeshBdrAttributes(
+      *pmesh, mfem_ext::MakeBdrAttributeList<std::vector<int>>(*pmesh, bdr_assign_coef, mfem_ext::digitize::floor));
+
+  // we'll get the top blue corner
+  auto blue = stars_attr_set.getComplement({1});
+
+  // we want all the indices that composite the stripes
+  // Take the difference of stripes_attr_set - (stars_attr_set = 1)
+  auto red_white = stripes_attr_set.getDifference(blue);
+
+  // Recombine the stripes sans top corner with the top corner
+  auto flag = blue.getUnion(red_white);
+  // Now red = 1, white = 2, blue = 3
+  mfem_ext::AssignMeshElementAttributes(*pmesh, flag.toList());
+
+  visit.RegisterField("stripes", &stripes.gridFunc());
+  visit.RegisterField("stars", &stars.gridFunc());
+
+  visit.SetCycle(0);
+  visit.Save();
+
+  // Integration tests on boundaries
+  serac::FiniteElementState density(*pmesh, FiniteElementState::Options{.order      = 1,
+                                                                        .vector_dim = 1,
+                                                                        .coll       = {},
+                                                                        .ordering   = mfem::Ordering::byVDIM,
+                                                                        .name       = "density",
+                                                                        .alloc_gf   = true});
+
+  // First define PWConstantCoefficient that corresponds to red=1, white = 2, blue = 3
+  mfem::Vector elem_densities(3);
+  elem_densities[0] = 1.;
+  elem_densities[1] = 0.0;
+  elem_densities[2] = 0.0;
+  auto density_coef = mfem::PWConstCoefficient(elem_densities);
+
+  // integrate on the boundary of the bottom
+  mfem::ParFiniteElementSpace& fes = *density.gridFunc().ParFESpace();
+  mfem::ParGridFunction        one_gf(&fes);
+  one_gf = 1.;
+
+  mfem::Array<int> bottom_marker(3);
+  bottom_marker    = 0;
+  bottom_marker[1] = 1;
+
+  auto surface_density_coef = mfem_ext::SurfaceElementAttrCoefficient(*pmesh, density_coef);
```

Comment:
Sometimes it's useful to be able to get "domain" elements attributes from their associated surface (bdr elements). This coefficient does that, and if we move away from using coefficients, it would be useful to have this information available in some form.

Some examples of how this is used is if you specifically only wanted to integrate on the surface of bdr elements associated with volume elements with a specific attribute.